subcommand/lifecycle-sidecar/command.go

```diff
@@ -100,13 +103,12 @@ func (c *Command) Run(args []string) int {
 	//
 	// The loop will only exit when the Pod is shut down and we receive a SIGINT.
 	for {
-		for _, svc := range svcs {
-			err := c.consulClient.Agent().ServiceRegister(svc)
-			if err != nil {
-				logger.Error("failed to sync service", "id", svc.ID, "err", err)
-			} else {
-				logger.Info("successfully synced service", "id", svc.ID)
-			}
+		cmd := exec.Command(c.flagConsulLocation, "services", "register", c.flagServiceConfig)
+		err := cmd.Run()
```

Comment:
We should also capture stdout and strerr here and include it in the logs. Otherwise, if command fails, the error message will just be something like "exit code 1"
```suggestion
		output, err := cmd.CombinedOutput()
```

Reply:
Updated.